core/cas-server-core-tickets-api/src/main/java/org/apereo/cas/ticket/registry/AbstractTicketRegistry.java

```diff
@@ -58,14 +59,27 @@ public void addTicket(final Ticket ticket) throws Exception {
 
     @Override
     public Ticket getTicket(final String ticketId) {
-        return getTicket(ticketId, ticket -> {
+        Ticket returnTicket = getTicket(ticketId, ticket -> {
             if (ticket == null || ticket.isExpired()) {
-                LOGGER.debug("Ticket [{}] has expired and will be removed from the ticket registry", ticketId);
+                if (ticket == null) {
+                    LOGGER.debug("Ticket [{}] not found but will be removed from the ticket registry just in case", ticketId);
+                } else {
+                    val ticketAgeSeconds = ZonedDateTime.now(ticket.getExpirationPolicy().getClock()).toEpochSecond() - ticket.getCreationTime().toEpochSecond();
+                    LOGGER.debug("Ticket [{}] has expired according to policy [{}] after [{}] seconds and [{}] uses and will be removed from the ticket registry",
+                        ticketId, ticket.getExpirationPolicy().getClass().getName(), ticketAgeSeconds, ticket.getCountOfUses());
+                }
                 deleteSingleTicket(ticketId);
                 return false;
             }
             return true;
         });
+        if (returnTicket != null) {
+            val ticketAgeSeconds = ZonedDateTime.now(returnTicket.getExpirationPolicy().getClock()).toEpochSecond() - returnTicket.getCreationTime().toEpochSecond();
+            if (ticketAgeSeconds < -1) {
+                LOGGER.warn("Ticket created [{}] seconds in the future. Check time synchronization on all servers", ticketAgeSeconds * -1);
```

Comment:
```suggestion
                LOGGER.warn("Ticket created [{}] second(s) in the future. Check time synchronization on all CAS servers", ticketAgeSeconds * -1);
```

Reply:
The current check is only going to log this if the time is more than 1 second off but I went ahead and changed the message. I initially had the check at greater than zero but thought people might get annoyed at warnings that time was off by 1 second. Not sure what the right amount of allowance is for time being off but with 10 second default expiration on service tickets, it doesn't take much.